packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -506,15 +383,16 @@ defineProperties(Element.prototype, {
     },
     getElementsByClassName: {
         value(this: HTMLBodyElement): HTMLCollectionOf<Element> {
+            let filteredResults;
             const elements = arrayFromCollection(
                 elementGetElementsByClassName.apply(this, ArraySlice.call(arguments) as [string])
             );
 
-            const filteredResults = getFilteredNodeListQueryResult(
-                this,
-                elements,
-                ENABLE_HTML_COLLECTIONS_PATCH
-            );
+            if (featureFlags.ENABLE_HTML_COLLECTIONS_PATCH) {
+                filteredResults = getFilteredNodeListQueryResult(this, elements, true);
```

Comment:
this is confusing, `getElementsByClassName` returns collection, but this is nodelist.